developer-extension/src/panel/components/json.tsx

```diff
@@ -1,58 +1,375 @@
+  value: unknown
+  defaultCollapseLevel?: number
+  getMenuItemsForPath?: GetMenuItemsForPath
 }
 
-export function Json({ src, name, collapsed }: JsonProps) {
-  const colorScheme = useColorScheme()
-  return (
-    <>
-      <Global
-        styles={[
-          {
-            '.object-key-val': {
-              paddingTop: '1px !important',
-              paddingBottom: '1px !important',
-            },
-            '.variable-row': {
-              paddingTop: '1px !important',
-              paddingBottom: '1px !important',
-            },
-            '.object-key': {
-              letterSpacing: '0 !important',
-            },
-            '.array-key': {
-              letterSpacing: '0 !important',
-            },
-          },
-        ]}
-      />
-      <ReactJson
-        src={src}
-        collapsed={collapsed}
-        theme={colorScheme === 'dark' ? 'monokai' : 'bright:inverted'}
-        name={name}
-        displayDataTypes={false}
-        style={{
-          backgroundColor: 'transparent',
-          fontFamily: 'menlo, sans-serif', // same as devtools
-          fontSize: '11px', // same as devtools
-          // By default, mantine sets font-smoothing: antialiazed, which makes some font hard to
-          // read
+type GetMenuItemsForPath = (path: string, value: unknown) => ReactNode
+
+const LINE_HEIGHT = '20px'
+const INDENT = 18
+const COLORS = {
+  dark: {
+    null: 'gray',
+    // eslint-disable-next-line id-denylist
+    number: 'orange',
+    // eslint-disable-next-line id-denylist
+    string: 'yellow',
+    // eslint-disable-next-line id-denylist
+    boolean: 'violet.4',
+  },
+  light: {
+    null: 'gray',
+    // eslint-disable-next-line id-denylist
+    number: 'orange.7',
+    // eslint-disable-next-line id-denylist
+    string: 'yellow.9',
+    // eslint-disable-next-line id-denylist
+    boolean: 'violet.7',
+  },
+}
+
+export const JsonContext = createContext<{
+  defaultCollapseLevel: number
+  getMenuItemsForPath?: GetMenuItemsForPath
+} | null>(null)
+
+type JsonValueDescriptor =
+  | {
+      parentType: 'root'
+      value: unknown
+      depth: 0
+      path: ''
+    }
+  | {
+      parentType: 'array'
+      parentValue: unknown[]
+      value: unknown
+      path: string
+      depth: number
+    }
+  | {
+      parentType: 'object'
+      parentValue: object
+      value: unknown
+      path: string
+      depth: number
+      key: string
+    }
+
+export const Json = forwardRef(
+  (
+    { value, defaultCollapseLevel = Infinity, getMenuItemsForPath, ...boxProps }: JsonProps & BoxProps,
+    ref: ForwardedRef<HTMLDivElement | HTMLSpanElement>
+  ) => {
+    const theme = useMantineTheme()
+    return (
+      <Box
+        ref={
+          // setting a HTMLDivElement | HTMLSpanElement ref messes with TS types as the component prop is also div | span
+          ref as any
+        }
+        {...boxProps}
+        component={doesValueHasChildren(value) ? 'div' : 'span'}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why is this necessary?

Reply:
The value is displayed inline if it does not have children, ex:

![Screenshot 2023-09-07 at 11 36 23](https://github.com/DataDog/browser-sdk/assets/61787/c47f4bfa-1e52-4172-b0d6-c6b7b7d3cd81)

WDYT? Is it overkill?